sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py

```diff
@@ -450,23 +451,33 @@ def __call__(self, *args, **kwargs) -> [..., Union["Command", "Parallel"]]:
     @contextmanager
     def _resolve_local_code(self):
         """Create a Code object pointing to local code and yield it."""
-        if hasattr(self, "code"):
-            code = getattr(self, "code")
-            # Hack: when code not specified, we generated a file which contains
-            # COMPONENT_PLACEHOLDER as code
-            # This hack was introduced because job does not allow running component without a
-            # code, and we need to make sure when component updated some field(eg: description),
-            # the code remains the same. Benefit of using a constant code for all components
-            # without code is this will generate same code for anonymous components which
-            # enables component reuse
-            if code is None:
-                with tempfile.TemporaryDirectory() as tmp_dir:
+        if not hasattr(self, "code"):
+            raise ValueError(f"{self.__class__} does not have attribute code.")
+        code = getattr(self, "code")
+        if code is not None and os.path.isfile(code):
+            yield Code(base_path=self._base_path, path=code)
+        else:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                if code is None:
+                    # Hack: when code not specified, we generated a file which contains
+                    # COMPONENT_PLACEHOLDER as code
+                    # This hack was introduced because job does not allow running component without a
+                    # code, and we need to make sure when component updated some field(eg: description),
+                    # the code remains the same. Benefit of using a constant code for all components
+                    # without code is this will generate same code for anonymous components which
+                    # enables component reuse
                     code = Path(tmp_dir) / COMPONENT_PLACEHOLDER
                     with open(code, "w") as f:
                         f.write(COMPONENT_CODE_PLACEHOLDER)
                     yield Code(base_path=self._base_path, path=code)
-            else:
-                # call component.code.setter first in case there is a custom setter
-                yield Code(base_path=self._base_path, path=code)
-        else:
-            raise ValueError(f"{self.__class__} does not have attribute code.")
+                else:
+                    # copy to temp folder to filter potential __pycache__
+                    # note that the dst here is one level deeper than the temp folder,
+                    # when upload code asset, the URL contains one level higher than the file,
+                    # if we simply copy to temp folder, the URL will contain random temp folder name,
+                    # that might result in unexpected issues and also break related tests.
+                    # therefore we copy deeper to avoid this break.
```

Comment:
maybe it's just a sanitizer issue. Given upload_hash is the same, although request uri changed, the same artifact will be returned. Whatever, we may resolve it in another PR

Reply:
yes, I have opened a task tracking this; may fix this in another PR related to resolving line separator